src/parser/jobs/smn/modules/Pets.js

```diff
@@ -288,50 +287,28 @@ export default class Pets extends Module {
 	output() {
 		const uptimeKeys = Array.from(this._petUptime.keys())
 
-		const data = {
-			labels: uptimeKeys.map(petId => this.getPetName(petId)),
-			datasets: [{
-				data: Array.from(this._petUptime.values()),
-				backgroundColor: uptimeKeys.map(petId => CHART_COLOURS[petId]),
-			}],
-		}
-
-		const options = {
-			responsive: false,
-			legend: {display: false},
-			tooltips: {enabled: false},
-		}
+		const data = uptimeKeys.map(id => {
+			const value = this._petUptime.get(id)
+			return {
+				value,
+				label: this.getPetName(id),
+				backgroundColor: CHART_COLOURS[id],
+				additional: [
+					this.parser.formatDuration(value),
+					this.getPetUptimePercent(id) + '%',
+				],
+			}
+		})
 
-		return <>
-			<div className={styles.chartWrapper}>
-				<PieChart
-					data={data}
-					options={options}
-					width={100}
-					height={100}
-				/>
-			</div>
-			<table className={styles.table}>
-				<thead>
-					<tr>
-						<th></th>
-						<th>Pet</th>
-						<th>Uptime</th>
-						<th>%</th>
-					</tr>
-				</thead>
-				<tbody>
-					{uptimeKeys.map(petId => <tr key={petId}>
-						<td><span
-							className={styles.swatch}
-							style={{backgroundColor: CHART_COLOURS[petId]}}
-						/></td>
-						<td>{this.getPetName(petId)}</td>
-						<td>{this.parser.formatDuration(this._petUptime.get(petId))}</td>
-						<td>{this.getPetUptimePercent(petId)}%</td>
-					</tr>)}
-				</tbody>
-			</table>
-		</>
+		return <PieChartWithLegend
+			headers={{
```

Comment:
Note that this prevents `PureComponent` from working; `headers` will never be `===` the previous update's `headers`.